### DISCUSSION TITLE
ESP32 alternative for pyb.USB_VCP

### DISCUSSION THREAD

**Discussion by jjscout** [micropython/micropython#15613] [ESP32]
I currently have a system that uses an STM processor that sends data to a host PC using a VCP over USB channel
The next hardware iteration uses an ESP32-S3 instead
The `pyb` library is not available on ESP32 and is oing to be deprecated so I ned an alternative implementation using the `machine` library, however i haven't found one yet.
What alternatives are available?
Thanks

**Comment by andrewleech** [comment]
There's work going on to migrate esp32 S2 & S3 to the micropython common tinyusb integration https://github.com/micropython/micropython/pull/15108
With that working it should also bring in compatibility with [machine.USBDevice](https://docs.micropython.org/en/latest/library/machine.USBDevice.html) which allows dynamic definition of the USB endpoints, eg. a second serial port.

If you instead mean you disconnect the first USB vcp port from repl using dupterm so you can use it for custom communications, well I'm planning on adding that to the common tinyusb integration but that hasn't been started yet.
